leapp/workflows/__init__.py

```diff
@@ -332,6 +337,19 @@ def run(self, context=None, until_phase=None, until_actor=None, skip_phases_unti
                     display_status_current_actor(actor, designation=designation)
                     current_logger.info("Executing actor {actor} {designation}".format(designation=designation,
                                                                                        actor=actor.name))
+
+                    metadata = actor.discover().copy()
+                    metadata.update({
+                        'consumes': [model.__name__ for model in metadata.get('consumes', ())],
+                        'produces': [model.__name__ for model in metadata.get('produces', ())],
+                        'tags': [tag.__name__ for tag in metadata.get('tags', ())],
+                        'dialogs': [dialog.serialize() for dialog in metadata.get('dialogs', ())],
+                        'phase': phase[0].name,
+                    })
+                    md = Metadata(kind='actor', name=actor.name, context=context,
+                                  hostname=os.environ['LEAPP_HOSTNAME'], metadata=metadata)
+                    md.store()
+
```

Comment:
@dkubek IIUC, this would be problematic. It stores the metadata about actors in time they are are going to be executed, instead of time when they are discovered. With that, we could just say what actors have been executed, but cannot say for sure whether all actors have been already executed.

I am also thinking yet about dialogs, as now they are hidden between all metadata. However, we would like to be able to work with dialogs in future in reports (or something like taht), so they could be shared with webUI applications (RH Insights, cockpit, foreman/satellite). I need to check what exact information is stored in this case now, but I can imagine we would like to have stored dialoges also somewhere else. My idea is to have a possibility to get list of produced dialogs. In this case, if I get it right, we have information about all dialogs defined in actors, but we will need information about actually asked dialogs.

I need to play around a little bit yet. But sharing my current ideas already.

Reply:
@pirat89 I was thinking about the point where should I save the actor. It seemed kind of unnecessary to store the information about actors which are not being executed and just have the information about what _has_ been executed. However, I think I now see that having even the information about discovered actors is valuable for debugging. And the information about executed actors can be always found in the audit. Will rework it.

As for the dialogs, I am aware that there is information missing and I have actually not come tome implementing that yet. So far I've been just puzzling over how they work so it's still WIP

Reply:
resolved! \o/